src/main/java/Parser.java

```diff
@@ -0,0 +1,85 @@
+public class Parser {
+    public static void addOrCompleteTask(String input, Task[] tasks){
+        String[] inputData = input.split(" ");
+        String[] taskDetails;
+        switch (inputData[0]) {
+        case "done": {
+            int completedTask = Integer.parseInt(input.split(" ")[1]);
+            completedTask--;
+            tasks[completedTask].markAsDone();
+            System.out.println("The Force is with you! The following task has been marked as done:");
+            System.out.println(tasks[completedTask]);
+            System.out.println("You still have " + Task.getTaskCounter() + " tasks in your list!");
+            UI.showDivider();
+
+            break;
+            }
+        case "deadline": {
+            taskDetails = findTaskDetails(inputData);
+            tasks[Task.getTaskCounter()] = new Deadline(taskDetails[0], taskDetails[1]);
+            System.out.println("Roger Roger. The following task has been added:");
+            System.out.println(tasks[Task.getTaskCounter()]);
+            Task.incrementTaskCounter();
+            System.out.println("You now have " + Task.getTaskCounter() + " tasks to complete in your list!");
+            UI.showDivider();
+
+
+            break;
+            }
+        case "event": {
+            taskDetails = findTaskDetails(inputData);
+            tasks[Task.getTaskCounter()] = new Event(taskDetails[0], taskDetails[1]);
+            System.out.println("Roger Roger. The following task has been added:");
+            System.out.println(tasks[Task.getTaskCounter()]);
+            Task.incrementTaskCounter();
+            System.out.println("You now have " + Task.getTaskCounter() + " tasks to complete in your list!");
+            UI.showDivider();
+
+
+            break;
+            }
+        case "todo": {
+            taskDetails = findTaskDetails(inputData);
+            tasks[Task.getTaskCounter()] = new Todo(taskDetails[0]);
+            System.out.println("Roger Roger. The following task has been added:");
+            System.out.println(tasks[Task.getTaskCounter()]);
+            Task.incrementTaskCounter();
+            System.out.println("You now have " + Task.getTaskCounter() + " tasks to complete in your list!");
+            UI.showDivider();
+
+
+            break;
```

Comment:
I believe a "default" case should be added here, so as to adhere to the coding standard on using a switch statement.

Reply:
You're right! Missed that one - thanks!